test/test_images/sender/main.go

```diff
@@ -129,15 +119,18 @@ func writeTerminationMessage(result interface{}) error {
 
 // isRetryable determines if the err is an error which is retryable
 func isRetryable(err error) bool {
-	var httpResult *cehttp.Result
-	if errors.As(err, &httpResult) {
+	var result *cehttp.Result
+	if errors.As(err, &result) {
 		// Potentially retry when:
 		// - 404 Not Found
 		// - 500 Internal Server Error, it is currently for reducing flakiness caused by Workload Identity credential sync up.
 		// We should remove it after https://github.com/google/knative-gcp/issues/1058 lands, as 500 error may indicate bugs in our code.
 		// - 503 Service Unavailable (with or without Retry-After) (IGNORE Retry-After)
-		sc := httpResult.StatusCode
-		return sc == 404 || sc == 500 || sc == 503
+		sc := result.StatusCode
+		if sc == 404 || sc == 500 || sc == 503 {
+			log.Printf("got error: %s, retry sending event. \n", result.Error())
+			return true
+		}
```

Comment:
Did you verify this manually?

Reply:
Yes, by inducing an error returned from processor.